### DISCUSSION TITLE
Change license to GPL

### DISCUSSION THREAD

**Discussion by kripper** [RosettaDrone/rosettadrone#176] [General]
@The1only Can we switch the license to GPL? This way, derived software products and source code changes based on Rosetta Drone would require other people to open source their code/changes too. It may help to get more pull requests and have more development of modules compatible with Rosetta Drone.

**Comment by The1only** [comment]
We could but there are many catches with GPL, and only if you sell a product that contains this code you will have to release the source, to customers…  and whom would fight it in court?Not sure it would help ! And I am for freedom.Best regardTerje Nilsen9Tek ASNorwayOn 15 Aug 2023, at 23:34, Christopher Pereira ***@***.***> wrote:﻿
@The1only Can we switch the license to GPL? This way, derived software products and source code changes based on Rosetta Drone would require other people to open source their code/changes too. It may help to get more pull requests and have more development of modules compatible with Rosetta Drone.

**Comment by kripper** [comment]
> We could but there are many catches with GPL, and only if you sell a product that contains this code you will have to release the source, to customers… 

I know what you mean and I always prefered BSD myself. But:

1. You could still keep a BSD license for **your own** derived closed source products.
2. Future developers would have to release their derived software products / source code changes and improvements. Otherwise, they may release closed source apps based on Rosetta Drone without even telling us :-(
